### ISSUE TITLE
Unable to optionally cast to array in SQL Editor mode

### ISSUE THREAD

**Issue by jacobsmith** [metabase/metabase#3657]
We have a few queries where users want to be able to add in multiple conditions (i.e., ID is in (`1, 2, 3`) or type is (`Ball, Bat`)).

I've been able to restructure a text variable as an array using type casts in Postgres, 
`select * from foobars where foobars.id in (string_to_array({{foobar_id}}, ',')::integer[])`,
However, as soon as I try and make that optional;

`select * from foobars [[ where foobars.id in (string_to_array({{foobar_id}}, ',')::integer[]) ]]`

I start getting syntax errors. I imagine that it has to do with the coercion to an array including the `[` and `]` characters which may be running amok of the variable parsing, but I don't know for sure.

Also, if there is a better way of doing this, I'd love to learn! This feels a bit hacky and like I might be overlooking a simpler solution.

Thanks!

**Comment by salsakran** [comment]
Can you copy+paste the actual error? That query looks right to me. 

```
select * from foobars [[ where foobars.id in (string_to_array({{foobar_id}}, ',')::integer[]) ]]
```

**Comment by jacobsmith** [comment]
Yes, the UI error is:

```
ERROR: syntax error at or near "[" Position: 156
```

And STDOUT is here (linking to a gist because it's ~100 lines of output) https://gist.github.com/jacobsmith/65145c74c2fa300f4e60e750faf84f07

**Comment by tlrobinson** [comment]
I'm pretty sure this regex is the issue: https://github.com/metabase/metabase/blob/8a113d8b5b83a7e172b25db5b28dfe522f4a9bc5/src/metabase/query_processor/sql_parameters.clj#L126

**Comment by camsaul** [comment]
Fixed by #4293
